src/main/java/wolox/training/models/Book.java

```diff
@@ -53,34 +55,51 @@ public void setGenre(String genre) {
     }
 
     public void setAuthor(String author) {
+        Preconditions.checkNotNull(author, ErrorConstats.FIELD_CANNOT_BE_EMPTY, "author");
```

Comment:
You can use this utility: `Strings.isNullOrEmpty(author)` that way you simplify these two lines into one (using `checkArgument`) 😄  
The same goes for the rest